src/twisted/mail/imap4.py

```diff
@@ -5691,7 +5691,7 @@ class Body:
         def __str__(self) -> str:
             return self.__bytes__().decode("ascii")
 
-        def __bytes__(self) -> bytes:
+        def getBytes(self, length: Optional[int] = None) -> bytes:
```

Comment:
Can we have docstring to this function?

This function is a long chain of if/elif ... commented out code ...

Now... maybe for an expert in the IMAP protocol, this makes perfect sense, and there is no need for extra documentation.

Reply:
I'm not entirely sure I understand all of it myself, but I've added my best effort at an explanation, let me know if it's not sufficient.